pkg/tenantControlNamespace/tenantControlNamespaceManager_test.go

```diff
@@ -0,0 +1,207 @@
+
+		_, ok := ns.Labels[discovery.TenantControlNamespaceLabel]
+		Expect(ok).To(BeTrue())
+	})
+
+	It("Create Namespace Twice", func() {
+		ns, err := namespaceManager.CreateNamespace(clusterID)
+		Expect(err).To(BeNil())
+		Expect(ns).NotTo(BeNil())
+		Expect(strings.HasPrefix(ns.Name, "liqo-tenant-control-")).To(BeTrue())
+		Expect(ns.Labels).NotTo(BeNil())
+
+		_, ok := ns.Labels[discovery.TenantControlNamespaceLabel]
+		Expect(ok).To(BeTrue())
+	})
+
+	It("Get Namespace", func() {
+		ns, err := namespaceManager.GetNamespace(clusterID)
+		Expect(err).To(BeNil())
+		Expect(ns).NotTo(BeNil())
+		Expect(strings.HasPrefix(ns.Name, "liqo-tenant-control-")).To(BeTrue())
+		Expect(ns.Labels).NotTo(BeNil())
+
+		_, ok := ns.Labels[discovery.TenantControlNamespaceLabel]
+		Expect(ok).To(BeTrue())
+
+		ns, err = namespaceManager.GetNamespace("unknownId")
+		Expect(err).NotTo(BeNil())
+		Expect(ns).To(BeNil())
+	})
+
+	Context("Permission Management", func() {
+
+		var client kubernetes.Interface
+		var namespace *v1.Namespace
+		var clusterRoles []*rbacv1.ClusterRole
+		var cnt int = 0
+
+		BeforeEach(func() {
+			cnt += 1
+			clusterID = fmt.Sprintf("testPermission%v", cnt)
+			client = cluster.GetClient().Client()
+
+			cr := &rbacv1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "role1",
+				},
+			}
+			cr, err := client.RbacV1().ClusterRoles().Create(context.TODO(), cr, metav1.CreateOptions{})
+			Expect(err).To(BeNil())
+			Expect(cr).NotTo(BeNil())
+			clusterRoles = append(clusterRoles, cr)
+
+			cr = &rbacv1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "role2",
+				},
+			}
+			cr, err = client.RbacV1().ClusterRoles().Create(context.TODO(), cr, metav1.CreateOptions{})
+			Expect(err).To(BeNil())
+			Expect(cr).NotTo(BeNil())
+			clusterRoles = append(clusterRoles, cr)
+
+			namespace, err = namespaceManager.CreateNamespace(clusterID)
+			Expect(err).To(BeNil())
+			Expect(namespace).NotTo(BeNil())
+		})
+
+		AfterEach(func() {
+			for _, cr := range clusterRoles {
+				err := client.RbacV1().ClusterRoles().Delete(context.TODO(), cr.Name, metav1.DeleteOptions{})
+				Expect(err).To(BeNil())
+			}
+			clusterRoles = []*rbacv1.ClusterRole{}
+
+			err := client.CoreV1().Namespaces().Delete(context.TODO(), namespace.Name, metav1.DeleteOptions{})
+			Expect(err).To(BeNil())
+		})
+
+		Context("Single ClusterRole", func() {
+
+			var rb []*rbacv1.RoleBinding
+
+			It("Bind ClusterRole", func() {
+				var err error
+				rb, err = namespaceManager.BindClusterRoles(clusterID, clusterRoles[0])
+				Expect(err).To(BeNil())
+				Expect(rb).NotTo(BeNil())
+				Expect(len(rb)).To(Equal(1))
+				Expect(rb[0].Namespace).To(Equal(namespace.Name))
+				Expect(len(rb[0].Subjects)).To(Equal(1))
+				Expect(rb[0].Subjects[0].Kind).To(Equal(rbacv1.UserKind))
+				Expect(rb[0].Subjects[0].Name).To(Equal(clusterID))
+				Expect(rb[0].RoleRef.Kind).To(Equal("ClusterRole"))
+				Expect(rb[0].RoleRef.Name).To(Equal(clusterRoles[0].Name))
```

Comment:
Nit. This could probably be extracted to a function and reused also below to avoid code duplication